### ISSUE TITLE
140000 days to finish 10w epochs training???

### ISSUE THREAD

**Issue by PkuRainBow** [deepinsight/insightface#163]
@nttstar Sorry to bother you. I guess there must exist something I have missed. Could you help me?

In your implementations, we need to train for 10000 epoches(line 93 of train_softmax.py). 

`
parser.add_argument('--end-epoch', type=int, default=100000, help='training epoch size.')
`

But it seems that you mentioned that you train for 20w iterations rather than epoches in your paper !

Training for 10w epoches over 2000w images....... Crazy!!!!!!!

I have trained your methods with 4XP100 GPUs for 7 days and only finished 46 epoches   ......................

Each epoch takes about 10000 iterations as I use smaller batch size=64 on each GPU.

So 10w / 50 = 2000 x 7 days ..... 140000 days????

Here I provide some training log,

```
INFO:root:Epoch[46] Batch [4360]        Speed: 61.02 samples/sec        acc=0.925391
INFO:root:Epoch[46] Batch [4380]        Speed: 317.33 samples/sec       acc=0.923633
INFO:root:Epoch[46] Batch [4400]        Speed: 317.54 samples/sec       acc=0.924609
INFO:root:Epoch[46] Batch [4420]        Speed: 317.96 samples/sec       acc=0.922656
INFO:root:Epoch[46] Batch [4440]        Speed: 316.52 samples/sec       acc=0.928906
INFO:root:Epoch[46] Batch [4460]        Speed: 318.25 samples/sec       acc=0.923633
INFO:root:Epoch[46] Batch [4480]        Speed: 317.42 samples/sec       acc=0.924023
INFO:root:Epoch[46] Batch [4500]        Speed: 316.43 samples/sec       acc=0.922656
INFO:root:Epoch[46] Batch [4520]        Speed: 316.35 samples/sec       acc=0.924219
INFO:root:Epoch[46] Batch [4540]        Speed: 317.87 samples/sec       acc=0.927734
INFO:root:Epoch[46] Batch [4560]        Speed: 319.68 samples/sec       acc=0.931055
INFO:root:Epoch[46] Batch [4580]        Speed: 315.75 samples/sec       acc=0.926953
INFO:root:Epoch[46] Batch [4600]        Speed: 318.45 samples/sec       acc=0.923242
```

**Comment by MyraBaba** [comment]
no answer?

**Comment by nttstar** [comment]
ignore the `num_epochs` setting and stop the training manually until verification accuracy do not change for a while
